extensions/ql-vscode/src/view/model-editor/ModelEditor.tsx

```diff
@@ -281,6 +281,7 @@ export function ModelEditor({
       t: "switchMode",
       mode: newMode,
     });
+    setMethods([]);
```

Comment:
In practice this works just fine.

The only thing I'm wondering is if it's ok that the methods in `ModelEditor` and in `ModelEditorView` are getting temporarily out of sync. Another option could be that when handling the `switchMode` message, we set `ModelEditorView.methods = []` and emit a `setMethods` message. Ultimately it doesn't matter much because we then call `loadExternalApiUsages` which updates the methods.

In this case I think this is ok because the methods get recomputed immediately. But it is something to be aware of generally that we need to keep this state in sync.

Reply:
Good point, I think setting the `methods` on the `ModelEditorView` is better, so I'll update the PR to do that instead.